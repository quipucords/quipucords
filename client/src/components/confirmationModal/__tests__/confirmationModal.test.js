import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import ConfirmationModal from '../confirmationModal';

describe('ConfirmationModal Component', function() {
  const generateEmptyStore = () => configureMockStore()({ confirmationModal: {} });

  it('should shallow render a basic component', () => {
    const store = generateEmptyStore();
    const props = { show: true };
    const wrapper = shallow(<ConfirmationModal {...props} />, { context: { store } });

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
