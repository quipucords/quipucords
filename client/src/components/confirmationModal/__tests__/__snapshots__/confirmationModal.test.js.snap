// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal Component should shallow render a basic component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <button
      aria-hidden="true"
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <Icon
        name="close"
        type="pf"
      />
    </button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Confirm
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="confirm-modal-body"
    >
      <span
        className="confirm-modal-icon"
      >
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
      </span>
      <span
        className="confirm-modal-content"
      >
        <span
          className="spacer"
        />
        <div
          className="confirm-modal-content-heading"
        />
        <div />
        <span
          className="spacer"
        />
      </span>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      autoFocus={true}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      onClick={[Function]}
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
    >
      Confirm
    </Button>
  </ModalFooter>
</Modal>
`;
